cmd/sink/logs/logs.go

```diff
@@ -7,91 +7,80 @@ import (
 )
 
 const (
 	celPrefix        = "cel:"
 	containerNameCel = "cel:spec.containers.map(m, m.name)"
+	loggingDirEnvVar = "KUBEARCHIVE_LOGGING_DIR"
 )
 
-var (
-	configCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Name:      "kubearchive-config",
-		Namespace: "kubearchive",
-	}
-	loggingCmRef = corev1.ObjectReference{
-		Kind:      "ConfigMap",
-		Namespace: "kubearchive",
+func filesInDir(path string) (map[string]string, error) {
+	files := make(map[string]string)
+	dir, err := os.ReadDir(path)
+	if err != nil {
+		return files, err
 	}
-	loggingCmSelector = corev1.ConfigMapKeySelector{
-		Key: "logging-configmap-name",
+	for _, file := range dir {
+		// kubernetes will create symlinks that start with '..' that should be ignored
+		if !file.IsDir() && !strings.HasPrefix(file.Name(), "..") {
+			var filePath strings.Builder
+			fmt.Fprintf(&filePath, "%s/%s", path, file.Name())
+			files[file.Name()] = filePath.String()
+		}
 	}
-)
+	return files, nil
+}
 
-func getKubeArchiveLoggingCm(ctx context.Context, kubeClient kubernetes.Interface) *corev1.ConfigMap {
-	emptyCm := &corev1.ConfigMap{Data: map[string]string{}}
-	kaConfigCm, err := kubeClient.CoreV1().ConfigMaps(configCmRef.Namespace).Get(
-		ctx,
-		configCmRef.Name,
-		metav1.GetOptions{},
-	)
-	if err != nil {
-		slog.Warn(
-			"error while getting ConfigMap",
-			"ConfigMap",
-			fmt.Sprintf("%s/%s", configCmRef.Namespace, configCmRef.Name),
-			"error",
-			err.Error(),
-		)
-		return emptyCm
+func loggingConfigFromFiles(files map[string]string) (map[string]string, error) {
+	conf := make(map[string]string)
+	for name, path := range files {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return conf, err
+		} else {
+			conf[name] = strings.TrimSpace(string(data))
+		}
 	}
-	var exists bool
-	loggingCmRef.Name, exists = kaConfigCm.Data[loggingCmSelector.Key]
+	return conf, nil
+}
+
+func getKubeArchiveLoggingConfig() map[string]string {
+	loggingDir, exists := os.LookupEnv(loggingDirEnvVar)
 	if !exists {
-		slog.Warn("logging ConfigMap not provided", "key", loggingCmSelector.Key)
-		return emptyCm
+		slog.Info("Skipping logging configuration. Environment variable not set", "env var", loggingDirEnvVar)
+		return make(map[string]string)
 	}
-	loggingCm, err := kubeClient.CoreV1().ConfigMaps(loggingCmRef.Namespace).Get(
-		ctx,
-		loggingCmRef.Name,
-		metav1.GetOptions{},
-	)
+	files, err := filesInDir(loggingDir)
 	if err != nil {
-		slog.Warn(
-			"error while getting ConfigMap",
-			"ConfigMap",
-			fmt.Sprintf("%s/%s", loggingCmRef.Namespace, loggingCmRef.Name),
-			"error",
-			err.Error(),
-		)
-		return emptyCm
+		slog.Error("Could not read logging config", "path", loggingDirEnvVar, "error", err.Error())
+		return make(map[string]string)
 	}
-	return loggingCm
+	loggingConf, err := loggingConfigFromFiles(files)
+	if err != nil {
+		slog.Error("Could not get value for logging config", "error", err.Error())
+		return make(map[string]string)
```

Comment:
If an empty map is returned here, does that mean an attempt will be made to generate URLs?  I would assume we would return nil and no attempt would be made.

Reply:
I've changed how the return so it now returns a `nil` map and an error. PTAL